web3/manager.py

```diff
@@ -74,19 +74,18 @@ def default_middlewares(web3):
     # Provider requests and response
     #
     def _make_request(self, method, params):
-        if self.provider:
-            request_func = self.provider.request_func(self.web3, tuple(self.middleware_stack))
-            self.logger.debug("Making request. Method: %s", method)
-            return request_func(method, params)
-        else:
-            raise UnhandledRequest(
-                "No provider available to respond to the RPC request:\n"
-                "method:{0}\n"
-                "params:{1}\n".format(
-                    method,
-                    params,
-                )
-            )
+        request_func = self.provider.request_func(
+            self.web3,
+            tuple(self.middleware_stack))
+        self.logger.debug("Making request. Method: %s", method)
+        return request_func(method, params)
```

Comment:
@kclowes @njgheorghita
Before I merge this, it would be good to get a couple sets of eyes on this.  Its the only place in this PR that touches the critical api.

With no longer iterating over the provider list, I think the UnhandledRequest exception is unnecessary and we can let the CannotHandleRequest pass through.   Or maybe this should be catching the "CannotHandleRequest" error like before?

For reference here is the request function when it had the provider list so you dont have to dig through the history:

```
    #   
    # Provider requests and response
    #   
    def _make_request(self, method, params):
        for provider in self.providers:
            request_func = provider.request_func(self.web3, tuple(self.middleware_stack))
            self.logger.debug("Making request. Method: %s", method)
            try:
                return request_func(method, params)
            except CannotHandleRequest:
                continue
        else:
            raise UnhandledRequest(
                "No providers responded to the RPC request:\n"
                "method:{0}\n"
                "params:{1}\n".format(
                    method,
                    params,
                )   
            )   
```

Reply:
From what I understand, it looks good to me. I'm thinking it's fine to let the `CannotHandleRequest` exception pass through from the provider rather than catching it here

Reply:
Though, maybe beefing up the msg [here](https://github.com/ethereum/web3.py/blob/f7cfd4e0f56500913c14ee50f40034b0f2e61bb4/web3/providers/auto.py#L78) to have more info like in the `UnhandledRequest` msg from before could be useful

Reply:
Looks correct.

Reply:
If `CannotHandleRequest` is no longer used anywhere it can be removed (didn't look to see if you already did this)

Reply:
Yeah, I agree with @njgheorghita. `CannotHandleRequest` makes sense to me, especially with a better error message!